### DISCUSSION TITLE
Como defender o não suporte ao IE8? (com foco no $$$)

### DISCUSSION THREAD

**Discussion by renatogalvones** [frontendbr/forum#168] [Discussão]
Fala ladies e rapeize!

Estamos passando por um período bacana de mudanças aqui na Catho e uma das coisas que queremos tentar encaixar nesse período é que **deixemos de dar suporte ao IE8**.

Sabemos que simplesmente apresentar uma cacetada de argumentos técnicos não necessariamente conseguiríamos convencer "os caras" de que vale a pena deixar o IE8 para trás, pelo simples fato que: 

> **O IE8 ainda é lucrativo**.

A porcentagem, de fato, é bem baixa, mas quando o volume de pessoas é grande mesmo uma porcentagem baixa vira uma cifra significativa no caixa da empresa e aí que a gente se complica.

Então faço aqui algumas perguntas:
- Vocês já passaram por algo parecido na empresa de vocês?
- Vocês já encontraram algum material/estudo/levantamento onde algo parecido aconteceu?
- Vocês ainda dão suporte ao IE8 na empresa de vocês?
- Que argumentos vocês utilizariam se estivessem no nosso lugar?

Valeu!

**Comment by kazzkiq** [comment]
Cara, o segredo está em quais argumentos você usa e como você apresenta esses argumentos.

Geralmente o diretor (ou alto executivo) que bate o martelo é alguém que tem pouca (ou nenhuma) experiência em desenvolvimento. Pra essa pessoa, suportar ou não outro navegador é questão de "algumas linhas de código" e nada mais. Não importa se vai dar um pouco mais de trabalho, o que importa é que o lucro desses usuários antigos também cheguem ao caixa da empresa.

O executivo que dá a palavra final geralmente conhece apenas uma língua: a do dinheiro. O que é aceitável, afinal esse é o trabalho dele: tomar decisões que gerem dinheiro para empresa.

E o seu trabalho é explicar pra ele porque a decisão de suportar um navegador antigo vai **tirar dinheiro** da empresa.

Eis alguns argumentos que podem te ajudar:

#### 1. Prazo

Navegadores antigos são cheios de bugs e falhas de segurança, além de não suportarem novas tecnologias. Isso significa que um programador que leva minutos para gerar um código em navegadores novos, gasta **horas** ou **dias** para gerar o mesmo código em navegadores antigos. E quanto maior o sistema, maior a diferença entre os dois códigos, o que afeta o prazo de entrega do projeto, além de nos levar ao segundo problema:

#### 2. Mão de obra

Com um projeto grande tendo que suportar navegadores antigos, agora você possui basicamente dois sistemas: Um atual, que possui todas as funcionalidades para usuários com navegadores novos, e um cheio de gambiarras, de workarounds, e de código legado para conseguir fazer tudo rodar em navegadores antigos. Toda vez que alguma funcionalidade nova entrar no projeto, você terá trabalho em dobro para implementar para os dois casos (navegadores novos e antigos). E dificilmente o mesmo número de pessoas vai conseguir dar vazão a todo esse código extra. Isso sem falar do impacto extra na equipe de testes.

#### 3. Segurança

Como dito no primeiro item, navegadores obsoletos são [recheados de falhas de segurança](https://www.google.com.br/search?q=ie8+security+issues). E pra piorar, a Microsoft ainda informou que a partir deste ano [ela não vai mais corrigir problemas nos navegadores antigos](http://www.techtimes.com/articles/12659/20140811/dingdong-internet-explorer-8-is-dead-microsoft-will-end-its-life-in-january-2016.htm). Em outras palavras, até a dona dos navegadores desistiu de ficar corrigindo programas velhos e problemáticos de mais de 10 anos de idade. Mas se o executivo da sua empresa acha uma boa ideia suportar estes navegadores correndo risco de expor usuários do seu site à diversos ataques, tudo bem. Depois que a marca da empresa perder valor por causa de usuários reclamando na internet que tiveram as contas invadidas (mesmo que a culpa seja do navegador), ele não pode dizer que você não avisou. ;)

#### 4. Não-suporte à novas tecnologias.

Notificações? Geolocalização? WebGL? Animações 3D? Gráficos, efeitos visuais, e todas essas coisas legais que todo mundo ta usando hoje em dia na internet? Esquece! Toda vez que uma dessas novidades for proposta, o executivo vai ter que pensar em como disponibilizar essa mesma experiência para os navegadores antigos. E na maioria dos casos, a resposta vai ser: não tem como. Ou ele cria duas experiências totalmente diferentes de acordo com o navegador de cada usuário, ou ele nem tenta melhorar a experiência pra nenhum dos dois. Enquanto o projeto estiver preso à ~~uma ancora~~ um navegador antigo, fica difícil até mesmo modernizar a própria empresa. E ele com certeza sabe que no mundo dos negócios, quem não se reinventa e moderniza, morre.

#### 5. Usuários de navegadores antigos gastam na internet?

Esse item é mais um exercício de imaginação do que um fato concreto. Mas nós sabemos que fazem **mais de 7 anos** que saiu o último sistema operacional que possuía o IE8 por padrão. Todos os novos computadores vêm com navegadores novos. É praticamente impossível encontrar uma casa que possua um computador tão antigo hoje em dia (a casa dos seus avós e a do seu tio da roça não contam). Será que quem tem dinheiro para contratar o serviço da Catho (ou fazer qualquer outra compra na internet) realmente possui um computador de quase uma década de idade? Se a pessoa tem conhecimento e capacidade de efetuar uma compra online, quais as chances de ela possuir um navegador e/ou um computador tão antigo? Será que usuários de IE8, quando comparados com usuários de navegadores novos, conseguem acompanhar a taxa de conversão e de lucro? Eles realmente compram? Vale a pena gastar tanto dinheiro, recursos e tempo para dar suporte à um grupo que não da lucro? Fica aí a dúvida.

#### 6. Custo-benefício ZERO

Somando todos os itens acima, não é difícil imaginar que o custo para a manutenção de um projeto que suporte navegadores antigos fique basicamente inviável. Você sabe disso, a equipe de desenvolvimento sabe disso, mas o executivo que bate o martelo não sabe _ainda_. Quando ele entender que o gasto para suportar essa pequena porcentagem de navegadores antigos é MAIOR do que o lucro, tenho certeza que ele muda de opinião. ;)

Então o que eu acho é: Monte uma apresentação com todos os argumentos que você e a equipe conseguirem juntar, marque uma reunião com os responsáveis pela decisão final, e explique de forma simples, didática e direta o porque de não ser uma boa ideia, do ponto de vista financeiro, a decisão de continuar suportando navegadores antigos.

Artigos relacionados:
- https://www.smashingmagazine.com/2012/07/old-browsers-are-holding-back-the-web/
- https://www.smashingmagazine.com/2012/07/dear-web-user-please-upgrade-your-browser/
- http://money.aol.co.uk/2012/06/14/online-shop-taxes-buyers-using-an-old-browser/

**Comment by renatogalvones** [comment]
@kazzkiq Muito obrigado de verdade pelo seu tempo escrevendo esse post.

Basicamente, é tudo que já pensamos aqui e mais um pouco. A idéia que você deu é exatamente o que estamos fazendo: juntando todos os argumentos(focando sempre no $$$) pra apresentar para eles.

Valeu mesmo!

Mais alguém querendo participar?

@neysimoes passou por algo parecido?

**Comment by bernardodiasc** [comment]
Nos meus tempos de freelancer eu só dizia: "o seu projeto contém funcionalidades mais modernas do que o browser que voce quer consegue suportar. caso insista precisarei cobrar o dobro do preço. não recomendo de forma alguma." rsrsrs

**Comment by barbier** [comment]
"A porcentagem, de fato, é bem baixa, mas quando o volume de pessoas é grande mesmo **uma porcentagem baixa vira uma cifra significativa no caixa da empresa** e aí que a gente se complica."

Cara, sinto muito, mas acho que vocês vão precisar continuar dando suporte....
Eu trabalhei numa empresa em que os caras não largavam o osso por causa disso. Há uns anos, 5 ou 6, um amigo trabalhou numa grande empresa de e-commerce onde o IE6 gerava uma receita de R$1M.
Quando o argumento da empresa é que o browser x gera tanto de receita para ela, fica difícil argumentar contra....

**Comment by felipefialho** [comment]
Sensacional a resposta do @kazzkiq, :clap: :clap:  

No seu caso é um pouco complicado, já que trabalha com uma empresa que funciona com assinaturas e etc. Funciona mais ou menos como nos sites de e-commerce, talvez essa pequena % de pessoas usando IE8, signifiquem alguns milhares de reais a mais na conta da empresa todo mês.

Porém você pode (e deve) usar os argumentos do @kazzkiq, especialmente na questão de segurança, no fato da própria Microsoft ter parado de dar suporte e no tempo gasto para que as funcionalidades funcionem no IE8.

Outra coisa que pode funcionar, é verificar o **tempo gasto x retorno do lucro**, por exemplo, não sei qual o valor exato que essas pessoas que usam IE8 retornam para o site (só acesso não conta, tem que ver quantos assinam), mas você com certeza tem uma ideia do tempo gasto para implementar as funcionalidades em navegadores antigos. Sendo assim pega:
- O tempo gasto \* valor hora das pessoas envolvidas, e você tem um valor real do quanto custa desenvolver para IE8.
- Tenta levantar o quanto essas pessoas realmente trazem de lucro para a empresa

Sendo assim você teria um lucro (ou prejuízo) liquido, fica mais fácil discutir usando números.

É claro que em uma Amazon, 1% de usuários pode significar 1 milhão a menos de vendas. Mas vale a tentativa.

**Comment by renatogalvones** [comment]
@barbier Esse é nosso "medo"! Mas estamos na luta pra tentar convencer que dá pra ganhar mais grana com os browsers novos.

 @LFeh Então a grande treta é que estamos querendo implementar React + Redux mas se mantiver o suporte ao IE8 não vai ter como né? Mas sim, estamos levantando todos esse números e comparativos pra chegar com algo firmeza pros "caras". 💰 💰 💰 

Valeu a força rapaziada!

**Comment by lagden** [comment]
TL;DR

Não dá nem pra da suporte para IE9...

**Comment by lagden** [comment]
@renatogalvones 

Tive uma super idéia...
Deixa o site atual para `ie.seudomain.com`
E começa o novo...

Então se o pia acessar do IE você redireciona...
em 1 ano você mata `ie.seudomain.com`... Só acompanhar os stats!!!

**Comment by bernardodiasc** [comment]
Ou entao usa isso http://csswizardry.com/2013/04/shame-css/ :)

**Comment by robtrevizani** [comment]
Acho que uma abordagem interessante seria você fazer um comparativo entre os requisitos. Demonstrar que o tempo e dinheiro investidos no suporte ao IE8 se redirecionados, podem trazer maiores ganhos para empresa de imediato ou no médio/longo prazo.

Talvez não suportar o IE8 possa até significar um aumento no número de conversões e receita, uma vez que você estará disponibilizando seus recursos (mão de obra e infra) aos usuários mais propensos a tornarem-se clientes. Há que se verificar essas taxas de conversão.

É muito grande a quantidade de informação que pode ser extraída desse emaranhado de dados, tanto para refutar o suporte ao IE8 quanto para corroborar com ele. No final o que vai contar mesmo é a capacidade de influenciar os "caras" a tomar essa decisão.

**Comment by rubenmarcus** [comment]
Galera ,levantando esse tópico de novo: Porquê sites de bancos, tem que se dar suporte as vezes para IE6 e IE7, logo banco, os caras não tem grana pra investir internamente e externa para deixar mais modernos?

Eu sonho o dia que o Bankline do Bradesco será igual o site do Nubank. Mesmo sendo ruim,ainda é melhor que o do Santander.

**Comment by adeonir** [comment]
No caso dos bancos acho mais complicado ainda. O alcance é maior, **aquele computador tão antigo hoje em dia (a casa dos seus avós e a do seu tio da roça)** contam, pois qualquer um pode ter uma conta corrente/poupança.
Eles investem nos módulos de segurança Warsaw e Java.

**Comment by rubenmarcus** [comment]
Mas o banco tem como fazer um site moderno e uma versão antiga... Eu fui abrir o site do Nubank no IE8, nem abre

**Comment by rubenmarcus** [comment]
Banco Original no IE8:

Access Denied
You don't have permission to access "http://www.original.com.br/" on this server.
Reference #18.1c9f9fc9.1484829881.32fb6200

**Comment by rubenmarcus** [comment]
Eu não vejo muito sentido, sei que tem a solução de duas versões, mas sua avó ou tio na roça , geralmente prefere ir na agência. Sei disso porque moro com a minha vó, e ela gosta de ter tudo no papel,e conferir tudo no caixa, Online pra ela parece meio suspeito, como se fosse um "fantasma" , misterioso

Reativei essa discussão, porque to debugando o site da Printi no IE aqui e corresponde cerca de 0.3% das vendas.. Coloquei o https://browser-update.org/update-browser.html ..

É um saco ficar dando suporte mas fazer o que..

Tirando banco, e e-commerces (na Centauro tambem davamos suporte ) não vejo muito porque dar suporte pro IE..

**Comment by renatogalvones** [comment]
@rubenmarcus a treta aqui é que ainda trabalhamos com muitas empresas que ainda estão no Windows XP (leia-se, IE 8) e não dá pra simplesmente "matar" esses caras porque eles não vão se atualizar em breve e não tem como irmos até os caras fazer um treinamento de como usar um browser mais novo... rs

Essa parte do nosso trampo é bem complexa. =|

**Comment by rubenmarcus** [comment]
@renatogalvones Eu entendo completamente. Fiz um freela pro Itaú por uma agência, E o gerente da agencia falando com o cliente se pegava em frases como:

-"Testa essa tela no Chrome"
-Chrome e Firefox OK
-"Agora no IE'
-Bem no IE7 e 8 está quebrado o formulário..

Ai lá ia mais 2 dias ou 3 pra se arrumar a tela.. 

Eu so levantei o topico de novo exatamente por isso, eu não entendo porque essas empresas não se atualizam,o calculo de custo x beneficio deve valer a pena.

**Comment by renatogalvones** [comment]
Sempre, ou quase sempre, o peso da balança pro lado da grana vai ganhar.

Acho que o grande lance nesses casos é saber a hora certa de parar de dar suporte completamente sem comprometer muito os lucros e com boa visibilidade de retorno no futuro oriundos de uma aplicação boa e bem estruturada, de fácil manutenção, deploy, etc... que vai te permitir se diferenciar dos concorrentes.
